datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRoleAssignmentController.java

```diff
@@ -0,0 +1,136 @@
+package org.datavaultplatform.webapp.controllers.admin;
+
+import org.datavaultplatform.common.model.RoleAssignment;
+import org.datavaultplatform.common.model.RoleModel;
+import org.datavaultplatform.common.model.RoleType;
+import org.datavaultplatform.common.model.User;
+import org.datavaultplatform.webapp.exception.EntityNotFoundException;
+import org.datavaultplatform.webapp.services.RestService;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@Controller
+public class AdminRoleAssignmentController {
+
+    private final RestService rest;
+
+    @Autowired
+    public AdminRoleAssignmentController(RestService rest) {
+        this.rest = rest;
+    }
+
+    @PostMapping("/security/roles/{roleType}/{target}/user/update")
+    @PreAuthorize("hasPermission(#targetId, #type, 'MANAGE_ROLES')")
+    public ResponseEntity updateRoleAssignment(
+            @PathVariable("roleType") String type,
+            @PathVariable("target") String targetId,
+            @Valid @NotNull @RequestParam("assignment") Long assignmentId,
+            @Valid RoleAssignmentRequest assignmentRequest) {
+
+        RoleAssignment roleAssignment = rest.getRoleAssignment(assignmentId)
+                .orElseThrow(() -> new EntityNotFoundException(RoleAssignment.class, String.valueOf(assignmentId)));
+
+        User user = rest.getUser(assignmentRequest.user);
+        RoleModel role = rest.getRole(assignmentRequest.role).orElseThrow
+                (()  -> new EntityNotFoundException(RoleModel.class, String.valueOf(assignmentRequest.role)));
+
+        roleAssignment.setUser(user);
+        roleAssignment.setRole(role);
+
+        return ResponseEntity.ok().build();
+    }
+
+    @PostMapping("/security/roles/{roleType}/{target}/user/delete")
+    @PreAuthorize("hasPermission(#targetId, #type, 'MANAGE_ROLES')")
+    public ResponseEntity removeRoleAssignment(@PathVariable("roleType") String type,
+                                               @PathVariable("target") String targetId,
+                                               @RequestParam("assignment") long assignmentId) {
+
+        RoleAssignment assignment = rest.getRoleAssignment(assignmentId)
+                .orElseThrow(() -> new EntityNotFoundException(RoleAssignment.class, String.valueOf(assignmentId)));
+
+        RoleModel role = assignment.getRole();
+
+        if (role.getAssignedUserCount() > 0) {
```

Comment:
I don't understand why this check is needed given that this is the method that removes the role assignment and therefore reduces the assigned user count